Ability to decrypt kms encrypted values

achinthagunasekara opened the pull request:
#### SUMMARY

Adding ability to decrypt values encrypted with AWS KMS. (and encrypt values with AWS KMS)

in `my_role/defaults/main.yaml` value can be included that is encrypted with AWS KMS.

```
encrypted_value: >-
  {{ abcd123abcd13 abcd123abcd13abcd12abcd13
  abcd123abcd13abcd123abcd13abcd123ab1d1312
  abcd123abcd13abcd123abcd13abcd123ssdsas12
  sadsdsasd | kms_decrypt(KEY_ARN) }}
```

lchantre committed:
Update filesystem.py

add ocfs2 support for module filesystem
Update setup.yml
delete trailing spaces
add ocsfs2 defaut var
Install ocfs2-tools for all linux
Testing ocfs2 on for Ubuntu - restrict blkid to be be done

Commit:
remove bad lines

add new line and update choice documentation
adding description for ocfs2 support
use correct variable ansible_distribution to test ocfs2 with Debian
distribution
use ansible_os_family for Debian
increase ocfs2 fs size to 20M (minimal size 11 instead of 10M)

pilou- committed:
filesystem tests: update ocfs2 fs size on Trusty

Using Ubuntu 14.04, test fails because 'blkid' < 2.21 doesn't recognize
'ocfs2' filesystem smaller than 108Mo:
https://github.com/karelzak/util-linux/commit/6baa15039873aac5513abde621f60c63537b88e9

filesystem: fix MKFS_FORCE_FLAGS for ocfs2

mkfs.ocfs2 -F won't work because mkfs.ocfs2 asks for a confirmation:

    $ mkfs.ocfs2 -F img
    mkfs.ocfs2 1.6.4
    Cluster stack: classic o2cb
    Overwriting existing ocfs2 partition.
    WARNING: Cluster check disabled.
    Proceed (y/N):

The undocumented 'x' switch must be used too.

abadger committed:
Emit a warning when src is used with a state where it should be ignored

abadger committed:
Correct bug reporting the diff when state=touch as absent

Commit:
Improved scaleset facts (#39722)

* improved scaleset facts

* scaleset facts fix some errors

* adding version_added for format param

* trying to break lines

* fixed syntax

* small code restructuring

* fix syntax

* fixed now

* add new test to scaleset / scaleset facts

* make scaleset test more clear

* temporarily comment out...

* try to retrieve scaleset facts

* try to add postfix

* fixed mistake

* fixed problem when no loadbalancer attached

* fixed another bug

* fixed sanity and a few other

* fixed pep8

* another try

* changed ansible to curated

* updated tests

* updated example and a few other mods

* small fixes

* removed unnecessary pass

* removed some items from ignore.txt

* remove file added by mistake

mkrizek committed:
Clarify in docs that end_play affects all hosts (#40235)

thadguidry committed:
small grammar improvements to add clarity (#40157)

gdpak committed:
Update ROADMAP_2_6.rst

achinthagunasekara committed:
Fixing pylint issues

achinthagunasekara committed:
Fixing pylint issues

achinthagunasekara committed:
Ability to encrypt with KMS

achinthagunasekara committed:
Ability to encrypt with KMS

achinthagunasekara committed:
Ability to decrypt kms encrypted values

achinthagunasekara committed:
Ability to decrypt kms encrypted values

achinthagunasekara committed:
Swapping the args

achinthagunasekara committed:
Swapping the args

rvalle committed:
opennebula: new module one_host (#40041)

achaussier committed:
Fix flatten filter documentation

pfarmer committed:
cloudstack: added fetch_list=True where appropriate (#40233)

resmo committed:
cs_instance_facts: add host return (#40307)

ganeshrn committed:
Update ROADMAP_2_6.rst

ganeshrn committed:
Update ROADMAP_2_6.rst

suacide24 committed:
changed accportgrp to accbundle (#40104)

* changed accportgrp to accbundle

* fixed shippable complaints

* changed policy_group_type to interface_type

ganeshrn committed:
Add netconf_get module (#39869)

* Add netconf_get module

Implements part-1 of proposal #104
https://github.com/ansible/proposals/issues/104

*  Add netconf_get module
*  Refactor `get`, `get_config`, `lock`, `unlock`
   and `discard_changes` netconf plugin api's
*  Add netconf module_utils file which netconf module
   related common functions
*  Refactor junos and iosxr netconf plugins

* Fix source option handling

* Fix review comments

* Update botmeta file

* Update review comments and add support for lock

* Lock update fix

* Fix CI issue

* Add integration test and minor fixes

* Fix review comments

* Fix CI failure

* Fix CI issues

* Fix CI issues

* Fix review comments and update integration test

* Fix review comments

* Fix review comments

* Fix review comments

Fix reveiw comments

Commit:
Modifying cnos-facts, cnos_command and cnos-config in line with the design followed in Ansible. Adding unit test cases for these modules. Added plugins to support them. (#39955)

* Modifying cnos-facts, cnos_command and cnos-config in line with the design followed in Ansible. Adding unit test cases for these modules. Added plugins to support them.

* Removing doc fragment conflicts with other modules

* Replacing show with display

mattclay committed:
Add cnos to network target prefixes.

abadger committed:
Reorganize ensure_directory to make things clearer

* value of file_args['path'] is now more clear
* We already have b_path so reuser that
* Add comments

mikewiebe committed:
Fix legacy Nexus 3k integration test and module issues. (#40322)

* Add setup ignore_errors for nxos_config test

* Fix parse_fan_info for n3k

* Skip bidir tests for N3k

* Omit vni config for n3k

* Skip unsupported nxos_vrf_af test on N3K

* Add legacy N3K platform tag

abadger committed:
Rename original_basename parameter in various file-related modules

original_basename is a private internal parameter so name it
_original_basename to reflect that it is internal.

bcoca committed:
User unexpire (#39758)

* Allow negative values to expires to unexpire a user

Fixes #20096

(cherry picked from commit 34f8080a19c09cd20ec9c045fca1e37ef74bb1e6)
(cherry picked from commit 54619f70f4b79f121c5062d54e9732d3cbb24377)
(cherry picked from commit 8c2fae27d6e2af810112032bb1dfef5459035b7e)
(cherry picked from commit db1a32f8caa8c8b9f989baa65784d4b2b5cad1f8)

* tweaked and normalized

 - also added tests, made checking resilient

ryansb committed:
[AWS] Add Ansible Version to botocore user agent string (#39993)

Pull `config` out if it was provided to boto3_conn and add the useragent string

sijis committed:
docs: Suggest get_url or uri for file checksum validation (#40087)

Fixes #13665

bengerman13 committed:
allow non-integer values in /etc/default/passwd to mirror solaris behavior

dagwieers committed:
uri: Add form-urlencoded support to body_format (#37188)

* uri: Add form-urlencoded support to body_format

This PR adds form-urlencoded support so the user does not need to take
care of correctly encode input and have the same convenience as using
JSON.

This fixes #37182

* Various fixes

* Undo documentation improvements

No longer my problem

* Fix the remaining review comments

albers committed:
Add release date for v2.5.0 to changelog (#40321)

According to https://github.com/ansible/ansible/blob/stable-2.5/changelogs/CHANGELOG-v2.5.rst#v250
Ansible 2.5.0 was released on 2018-03-22.

Signed-off-by: Harald Albers <github@albersweb.de>

s-hertel committed:
Fix failing aws_ses_identity integration tests (#39560)

* Fix failing aws_ses_identity integration tests

Reduce boilerplate with yaml anchor

* remove unstable test alias

* Update feedback forwarding check to use desired state rather than
repeated API calls.

marwatk committed:
New Module: gitlab_deploy_key and related tests (#40097)

* Added module gitlab_deploy_key and related tests

* Fix sanity check issues

* Refactor to use common util method, add check_mode support

* Fix module shebang

marwatk committed:
New Module: gitlab_hooks module and related tests (#40096)

* Added gitlab_hooks module and related tests

* Fix sanity check issues

* Refactor to use common util method, add check_mode support

* Fix module shebang

samdoran committed:
Remove python-keyczar dependency (#38534)

* Remove python-keyczar dependency

* Remove python-keyczare from Dockerfiles

* Revert "Remove python-keyczare from Dockerfiles"

This reverts commit 264be66ad8337c5ccd0db76dc7000015500b6074.

samdoran committed:
Add better error messages and checking to known_hosts (#38307)

samdoran committed:
Do not join flag parameters in iptables module (#36658)

* Do not join flag parameters

This put a comma between every character of the tcp flag parameters, resulting in a bad iptables command.

Fixes #36490

* Use suboptions to ensure tcp_flags options are lists

* Add unit tests for tcp_flags

* Add example of how to use tcp_flags

abadger committed:
original basename minus src

samdoran committed:
Change behavior to behaviour in docs and tests (#39392)

* Change behavior to behaviour

- use existing fact to get hash setting rather than shell task
- fix code highlighting syntax in playbooks_variables.rst

* Re-wrote intro section; this entire topic needs a clean-up/rewrite.

abadger committed:
Move rewriting of path earlier in the argument handling as other things depend on the path

mkrizek committed:
redhat_subscription: C locale for stdout scraping (#40165)

bengerman13 committed:
S3 versioned lifecycle (#40161)

* - add tests for s3_lifecycle
- fix a bug comparing transitions with different storage_types

* make s3_lifecycle work with boto3

* add noncurrent version lifecycle rules

abadger committed:
Fix python-2.4ism that was still present in basic.py's use of tempfile

abadger committed:
unarchive now passes the E322 test

maxamillion committed:
Handle incorrect data type in list lookup plugin (#35483)

handle incorrect data type in list lookup plugin
Fixes #35481
test to ensure that loops properly handle incorrect datatypes

Signed-off-by: Adam Miller <admiller@redhat.com>

rambleraptor committed:
Adding support for GCP Compute Target Https Proxys (#38908)

rambleraptor committed:
Adding support for GCP Compute Target Tcp Proxys (#38624)

rambleraptor committed:
Adding support for GCP Compute Target Http Proxys (#38622)

resmo committed:
cs_vpc: implement state=started (#40319)

gtema committed:
fixes issue 39472: (#40341)

With python 3.6 spwd.getspnam returns PermissionError instead of
KeyError if user does not have privileges

ruizink committed:
Add new OpenStack module for changing metadata of compute instances (#19113)

* Add new OpenStack module for changing metadata of compute instances

* Add missing RETURN documentation

* Fix API calls to match shade's implementation.

API was changed in the following PR (openstack-infra/shade):
https://review.openstack.org/#/c/319395

* Replace type by isinstance when type checking

* Add ANSIBLE_METADATA

Signed-off-by: Mário Santos <mario.rf.santos@gmail.com>

* Bump the ansible version

Signed-off-by: Mário Santos <mario.rf.santos@gmail.com>

* Add support for check_mode

Signed-off-by: Mário Santos <mario.rf.santos@gmail.com>

* Fix sanity tests

Signed-off-by: Mário Santos <mario.rf.santos@gmail.com>

* Force the meta parameter to be a string

Signed-off-by: Mário Santos <mario.rf.santos@gmail.com>

* Use short GPL license

Signed-off-by: Mário Santos <mario.rf.santos@gmail.com>

* Change meta argument to dict

Signed-off-by: Mário Santos <mario.rf.santos@gmail.com>

scottsb committed:
1Password lookup plugin (#37207)

* add pytest_cache to gitignore

* onepassword lookup plugin

* fix linter/style test complaints

* second pass at making pycodestyle happy

* use json module instead of jq

* update copyrights, license & version added

* fix python2 compatibility

* doh. fix spacing issue.

* use standard ansible exception

* remove potentially problematic stdin argument

* actually call assertion method

* add support for top-level fields

* make vault uuids pedantically consistent in fixture

* fix new style issues

* ability specify section & correct case handling

* improve error handling

* add onepassword_raw plugin

* Add maintainer info

* Move common code to module_utils/onepassword.py

* Load raw data JSON data for easier use in Ansible

* Put OnePass class back inside lookup plugin

There is no good place for sharing code across lookups currently.

* Remove debugging code in unit tests

* Patche proper module in raw unit tests

* Add changelog entry

Co-authored-by: Scott Buchanan <sbuchanan@ri.pn>

Qalthos committed:
Don't rewrite remote paths when remote is secretly local (#40259)

* Don't rewrite remote paths when remote is secretly local

* Remote overriding is configurable in connection

* Use c.DEFAULT_LOCAL_TMP instead of remote_tmp

* Remove remote_is_local from ConnectionBase

* remote_is_local->_remote_is_local

* Add warning signs to _remote_is_local and explanatory comments to its use

maxamillion committed:
sysvinit service module (#34962)

* sysvinit service module

Signed-off-by: Adam Miller <admiller@redhat.com>

Qalthos committed:
HTTP(S) API connection plugin (#39224)

* HTTPAPI connection

* Punt run_commands to cliconf or httpapi

* Fake enable_mode on eapi

* Pull changes to nxos

* Move load_config to edit_config for future-preparedness

* Don't fail on lldp disabled

* Re-enable check_rc on nxos' run_commands

* Reorganize nxos httpapi plugin for compatibility

* draft docs for connection: httpapi

* restores docs for connection:local for eapi

* Add _remote_is_local to httpapi

achinthagunasekara committed:
Fixing issues and adding encrypt ability

achinthagunasekara committed:
Fixing issues and adding encrypt ability

caphrim007 committed:
small fixes and additions to bigip_virtual_address (#40362)

* Updated parameter names to match updated features in BIGIP
* Added support for route domains

achinthagunasekara committed:
Bug fix

achinthagunasekara committed:
Bug fix

caphrim007 committed:
Adds the virtual argument to bigip_policy_rule (#40373)

This patch allows the module ot manage forwarding actions to virtual
servers in addition to the existing pools argument

mattclay commented on lib/ansible/plugins/filter/aws_kms.py:
```diff
@@ -0,0 +1,52 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type  # pylint: disable=invalid-name
+import base64
+import botocore
```
You'll need a try/catch around imports that are not from the Python standard library. See an example here:

https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/filter/json_query.py#L26

achinthagunasekara replied:
Thanks @mattclay I've updated my changes accordingly :)

achinthagunasekara committed:
Improvements

achinthagunasekara committed:
Improvements

achinthagunasekara committed:
Removing pylint line

achinthagunasekara committed:
Removing pylint line

mkrizek committed:
rhsm_repository: use C locale for out/err scraping (#40338)

ganeshrn committed:
Fix netconf get_config lock/unlock (#40378)

Lock/unlock the datastore that is given as input for
instead of default to `running`

gdpak committed:
rename network_get network_put modules to net_get net_put (#40381)

Commit:
Memset: initial module, integration tests, module_utils and supporting documentation. (#40071)

*  * Memset:
   * module_utils and associated documentation.
   * module to manage DNS zones.
   * integration tests to run against Memset's API.

* * memset.py:
  * remove import of requests from memset.py in favour of internal Ansible modules.
  * import necessary Ansible modules for request and response handling.
  * create a response object in order to retain a stable interface to memset_api_call from existing modules.
  * rework memset_api_call to remove requests.
* memset_zone.py:
  * improve short description to be more verbose.
  * ensure all imports directly follow documentation.
  * remove any reference to requests.
  * correct keyerror carried over from elsewhere.

* remove dependency on requests from documentation string

* disable integration tests until we have a cleanup method available

adrianlzt committed:
Fixes #29818: chkconfig command is not daemonized, uses run_command()

In the last commit I modified the code to run commands when they are
daemonized. But the execution of "chkconfig" is not daemonized so it
uses "self.module.run_command(cmd)".

This commit set the default localize to allow proper screen scraping of
chkconfig command.

Commit:
add maintainer information for Memset modules and module_utils (#40385)

gundalow committed:
removed_in: 2.6 is valid (#40399)

bcoca committed:
corrected version added

dminca committed:
format module values in gce_net module documentation (#40047)

* format code in gce_net module

The values passed to the module weren't being highlighted properly in
the documentation page.

As a result, use markup code format for the values provided in order to
be spotted easily on the page.

Signed-off-by: Daniel Andrei Minca <mandrei17@gmail.com>

* separate phrases

Signed-off-by: Daniel Andrei Minca <mandrei17@gmail.com>

* apply review fix for module options

According to http://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#formatting-functions

Signed-off-by: Daniel Andrei Minca <mandrei17@gmail.com>

mattclay committed:
Clarify wording on 2.6 roadmap schedule. (#40408)

* Clarify wording on 2.6 roadmap schedule.

* Remove redundant use of "Core".

caphrim007 committed:
Fixes password missing in bigip_device_trust (#40414)

The documentation did not match the code. This patch fixes the code
to correctly fallback the peer password if necessary

caphrim007 committed:
Adds wait parameters to bigiq applications (#40420)

The wait parameter allows the module to wait or not wait when
an application is created. By default the modules will wait.

mattclay committed:
Disable failing cloudstack tests.

carchi8py committed:
Add Na_ontap_broadcast_domain module.  (#39753)

* add first module

* fix case

* Create na_ontap

* Fix small pep8 errors

* Fixes for issues found by Amit

* fixes for amit

* fix doc

* get doc to compile

caphrim007 committed:
Adds remaining fqdn params to bigip_node (#40427)

This patch adds params for auto_populate, address type, up interval
and down interval params to bigip node

caphrim007 committed:
Adds the bigip_gtm_global module (#40430)

This module can be used to manage the global GTM settings on a BIGIP

mikewiebe committed:
Skip N35 and N3L platforms for nxos_evpn_global test (#40333)

Akasurde committed:
Correct doc for cache plugin

Section is a part of ini, so removing it as separate key.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

jaredhocutt committed:
Fix example in ipa_dnszone module (#40443)

The ipa_dnszone module had a typo in the examples where "ipa_dnsrecord"
was being used instead of "ipa_dnszone".

achinthagunasekara committed:
Typo correction

achinthagunasekara committed:
Typo correction

mkrizek committed:
rabbitmq_vhost: add integration tests (#40387)

* rabbitmq_vhost: add integration tests

* Use import_tasks instead of include

mkrizek committed:
rabbitmq_plugin: add integration tests (#40389)

* rabbitmq_plugin: add integration tests

* Use import_tasks instead of include

justjais committed:
Integration tests for Infoblox 2.5 modules (#40344)

* Update nios.py

* Update nios.py

* Update nios.py

* nios lookup errors out when there are no results #37970  Open	

Indentation failure issue resolved

* Returning empty list instead of None

In case of no results, res will be returned as an empty list instead of None (implementing ganeshrn comment)

* infoblox ipv6 support changes

* infoblox ipv6 support changes

* for fixing pep8 errors

* moving ipaddr check to utils

* adding ipv6addr check

* increasing space to resolve pep8 error

* modified the playbook examples to valid ones

* Update nios_network.py

* integration tests for nios 2.5 modules

* modification done in existing integration nios testcases

* dns_view nios module tc

* host_record nios module tc

* network nios module tc

* network_view nios module tc

* zone nios module tc

* changes to fix shippabe errors for PR 40344

* PR40344 shippable fix

* PR40344 shippable fix

* PR40344 shippable fix

* PR40344 shippable fix

* PR40344 shippable fix

* PR40344 shippable fix

* PR40344 shippable fix

* PR40344 shippable fix

* PR40344 shippable error fix

* 40344 shippable fix

* 40344 shippable fix

* 40344 shippable fix

* 40344 shippable fix

* 40344 shippable fix

* 40344 shippable fix

* PR40344 shippable error fix for block comment should start with '# '

bcoca committed:
use correct object in error

fixes #40453

acatton committed:
Add the ability to specify an install_dir to the gem module (#38195)

* Add the ability to specify an install_dir to the gem module

* Add GEM_HOME when installing a non-global gem

* Add tests for custom gem path

* Fix sanity tests

* Add changelog entry

* Rebase and add tests for incorrect options

Co-authored by: Antoine Catton <devel@antoine.catton.fr>

samdoran committed:
Fix ctrl+c in pause module and add tests (#40134)

* Fix all cases with pause and ctrl+c
 - naked:
	- pause:
 - with prompt
	- pause: prompt=hi
 - time wait
	- pause: seconds=60
 - time wait with prompt
	- pause: seconds=60 prompt=hi


Fixes #35372

* Use curses to control stdout
* Use curses to clear lines on interactive input
* Validate input for echo parameter and fail nicely if invalid
* Add integration tests for pause module using pexpect
* Use try except when trying to determine erase sequence to account for lack of TTY in containers in tests
* Improve output validation for regular paus test
* Accept two digit precision for pause length in test
* Check for seconds when seconds is specificed, minutes when minutes is specified
* Add test for no TTY mode

Co-authored by: Toshio Kuratomi <a.badger@gmail.com>
Co-authored by: Brian Coca <brian.coca+git@gmail.com>

samdoran committed:
Move summary to top for issue template (#40401)

Qalthos committed:
Fixing issues with httpapi (#40388)

* I seem to have forgotten the back half of tests

* Set http timeout from persistent_command_timeout

* Tweak URL generation and provide URL on error

* Push var_options to connection process

* Don't wait forever if coming from persistent

* Don't send the entire contents of variables to ansible-connection

resmo committed:
cloudstack: fix query tags by resourceid (#40482)

bcoca committed:
reset_connection is not run_once

bcoca committed:
remove extra )

sivel committed:
Group logic detecting user, ensure rsync_opts is a list if omitted. Fixes #40483

sweenu committed:
Explicit availability of 'null' timeout in expect (#40429)

abadger committed:
Remove the cherrypick script

We only needed it for migrating cherrypicks between the unified repo and
the ansible-modules-* repos.  Now that we aren't supporting 2.3, we no
longer need this script.

kustodian committed:
postgresql_user: set encrypted as default and fix empty password reporting changed (#36931)

* Set encrypted as default and fix empty password reporting changed

* Starting with Postgres 10 `UNENCRYPTED` passwords are removed and
because of that this module fails with the default `encrypted=no`.
Also encrypted passwords are suported since version 7.2
(https://www.postgresql.org/docs/7.2/static/sql-createuser.html) which
went EOL in 2007 and since 7.3 it is the default. Because of this it
makes a lot more sense to make `encrypted=yes` the default. This won't
break backward compatibility, the module would just update the user's
password in the DB in the hashed format and everything else will work
like before. It's also a security bad practice to store passwords in
plain text. fixes #25823
* There was also a bug with `encrypted=yes` and an empty password always
reported as changed.
* Improved documentation for `encrypted`/`password` parameters, and
removed some obsolete notes about passlib.

* Fix clearing user's password to work with all versions of Postgres

* Add tests for clearing the user password

* Fix documentation atfer rebase

* Add changelog fragment

jborean93 committed:
win_environment: Added explicit check for null, empty string with state=present (#40468)

* win_environment: Added explicit check for null, empty string with state=present

* Added changelog fragment

mattclay committed:
Remove tab check from validate-modules.

It is redundant with the pycodestyle W191 "indentation contains tabs" check.

Qalthos committed:
Ios test fixes (#40503)

* Return messages generated from edit_config to module

* This does not seem to work that way

* Change test IP addresses to not conflict with device config

nitzmahone committed:
2.6 changelog gen/version/root dir cleanup (#40421)

* patched in changelog gen stuff from stable-2.5
* Makefile updates
* release.py as single-source-of-truth
* Remove obsolete ansible-core-sitemap.xml file.
* Move ROADMAP.rst into README.rst.
* dynamic rpm changelog, zap old deb/rpm changelogs
* fix changelog in MANIFEST.in
* Remove obsolete hacking/update.sh script.
* Remove ref to deleted authors script.
* Remove ref to removed module-formatter script.
* Update headings to match script names.
* MANIFEST.in cleanup
* removed RELEASES.txt and versions.yml
* removed obsolete release generation playbook/bits (not used since 2.5)
* misc Makefile cleanup
* speculative changes to DEB versioning
* allow override of DEB_VERSION/DEB_RELEASE

jborean93 committed:
winrm: source user from options than remote_user (#40467)

* winrm: source user from options than remote_user

* fixed up mock for kerberos import

* Added changelog fragment

* get hostname from option as well

mattclay committed:
New release v2.6.0a1

nitzmahone committed:
fixed prerelease support in deb packaging

mattclay committed:
Update Ansible release version to 2.6.0dev0.

mattclay committed:
Add botmeta entry for lib/ansible/release.py.

caphrim007 committed:
Fixes bigiq doc error (#40517)

Corrects the return values for a bigiq module

caphrim007 committed:
Adds bigiq utility license module (#40518)

This module can be used to manage utility licenses on a BIG-IQ

cbenhagen committed:
digital_ocean_domain: fix do not fail if domain already is absent (#40446)

Commit:
fix nxos_system issues (#40347)

marns93 committed:
New module: heroku_collaborator (#39001)

dericcrago committed:
simplified `DistributedVirtualSwitch` namespacing (#40191)

bigmstone committed:
Adding slxos_l3_interface module (#39320)

* Adding slxos_l3_interface module

* Corrected return values

The returned IPv4 address has a prefix length, not a subnet mask

bigmstone committed:
Adding slxos_vlan module (#39197)

* Adding slxos_vlan module

Commit:
uses .txt for shared content (#40416)

Akasurde committed:
VMware: vmware_guest Documentation update (#38377)

This PR refactors vmware_guest module

- VMware: Documentation update
- Elaborate more on effects of state on virtual machine.
- Specify 'force' option usage.
- Formatting
- add review comments

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

stravassac committed:
VMware: Fix file operation to work with ESXi directly (#40148)

Now, file_operation module can be used with both vCenter and Standalone ESXi server.

itdependsnetworks committed:
create cisco type 5 filters (#39901)

rambleraptor committed:
Adding support for GCP Compute Target Ssl Proxys (#39033)

rambleraptor committed:
New module: GCP Compute Instance Group Managers (#38934)

rambleraptor committed:
New module: GCP Compute Url Maps (#38625)

rambleraptor committed:
New module: GCP Compute Target Pools (#38623)

rambleraptor committed:
New module: GCP Compute Instance Groups (#38617)

rambleraptor committed:
New module:  GCP Compute Ssl Certificates (#38620)

rambleraptor committed:
New module: GCP Compute Images (#38616)

rambleraptor committed:
New module: GCP Compute Global Forwarding Rules (#38615)

rambleraptor committed:
New module: GCP Compute Global Address (#38614)

rambleraptor committed:
New module: GCP Compute Forwarding Rules (#38613)

rambleraptor committed:
New module: GCP Compute Firewalls (#39032)

Qalthos committed:
Fix persistent timeout definitions (#40540)

ryansb committed:
[docs] Update `ec2` module to use wait_for_connection instead of wait_for (#40171)

ryansb committed:
Add changelog for new GCP modules (#40542)

Qalthos committed:
Search all assigned ipv6 addresses instead of just the first (#40533)

* Search all assigned ipv6 addresses instead of just the first

* Add test for idempotency with multiple ipv6 addresses assigned

samdoran committed:
EdgeOS module improvements (#39530)

* Update docs and use to_text on strings

* Add warning to use network_cli

cheese committed:
Make xenserver_facts actually work (#35821)

* Get the str value of xmlrpc.client.DateTime

* get_all_records should be used instead of get_all

* Facts returned with 'ansible_facts'

* Remove some redundant code

* Add cheese as maintainer

* Add changelog entry

mattclay committed:
Update 2.6 roadmap schedule.

caphrim007 committed:
Adds bigip_monitor_external module (#40558)

This module can be used to manage external LTM monitors on a bigip

caphrim007 committed:
Corrects docs for ltm external monitor (#40559)

Examples were incorrect. This fixes them

caphrim007 committed:
Adds the bigip_gtm_monitor_external module (#40560)

This module can be used to manage GTM external monitors

Commit:
Add flatpak module (#40286)

* Adding flatpak module

Includes contributions by dagwieers

* Incorporates feedback by dagwieers

* Improves and aligns documentation to conventions

* Makes matching for url more robust

Commit:
Add flatpak_remote module (#40189)

* Adding flatpack_remote module

Includes contributions by oolongbrothers: adding documentation fixes, fixing import placements, fix module idempotency, improving error propagation.

* Aligns module with conventions and best practices

For improved robustness and readability; based on feedback by
dagwieers.

* Improves the module documentation

* Aligns copyright statement to conventions

* Renames remote > flatpakrepo_url and reworks documentation

ekohl committed:
foreman: Fix get_option regression (#40534)

d303c0706c69831f4e5233a5c9d43b8f2db6e5ac introduced a regression because
of a partial port from _options to get_option. This fixes it.

jborean93 committed:
always_run: removed deprecated always_run task option (#40470)

jborean93 committed:
powershell: Respect remote_tmp path set by the user (#40210)

* powershell: Respect remote_tmp path set by the user

* Fixed up linting error and typo

* Added changelog

mattclay committed:
Fix shebangs and file modes and update tests. (#40563)

* Add execute bit sanity test and apply fixes.
* Add shebang test for `lib` dirs and apply fixes.
* Shebang and execute bit cleanup.

gnadaban committed:
Implemented support for Repository selection when installing module (#39909)

* Implemented support for Repository selection when installing module

* Fixed AllowClobber check

* Fixing missing AllowClobber argument

* Fixed missing -Force switch

* Test installing package from custom repo

* Added custom repo variables

* Fixed repo registration

* Modified output check

* Modified output check

* Adding debug output

* Modified update handling

* Fixed output check

* Added missing PowerShellGet 1.6.0 install

* Added error handling

* Modified test output check

* Fixing output check

* Fixing output filter

* Implemented registering custom repo

* Fixing registering custom repo

* Fixing registering custom repo

* Fixing registering custom repo

* Tweaked module to shorten execution time

* Removed installing PowerShellGet 1.6.0

* Re-added Install-NugetProvider

Commit:
Detect separate git dir and set git config path value appropriately

PR #38016

Co-authored-by: Sviatoslav Sydorenko <wk.cvs.github@sydorenko.org.ua>

jitran commented on lib/ansible/plugins/filter/aws_kms.py:
```diff
@@ -0,0 +1,70 @@
+
+def aws_kms_encrypt(plaintext, key_arn):
+    """ Encrypt with KMS """
+    if not HAS_DEPENDENCIES:
```
Hi Archie, is it worth moving this conditional block into an internal function as it's repeated twice?

achinthagunasekara replied:
@jitran Good point. Updated with proposed changes.

achinthagunasekara committed:
Moving repeated code into a function.

achinthagunasekara committed:
Moving repeated code into a function.

achinthagunasekara committed:
Spelling correction.

achinthagunasekara committed:
Spelling correction.

achinthagunasekara committed:
Update to comments

achinthagunasekara committed:
Update to comments

achinthagunasekara committed:
Merge https://github.com/ansible/ansible into devel

achinthagunasekara committed:
Merge branch 'devel' into dev-gunasekara-kms

achinthagunasekara committed:
Merge branch 'devel' into dev-gunasekara-kms

achinthagunasekara committed:
Rebase with devel

achinthagunasekara committed:
Merge branch 'dev-gunasekara-kms' of https://github.com/achinthagunasekara/ansible into dev-gunasekara-kms

achinthagunasekara committed:
Reverting changes back

achinthagunasekara commented:
Closing this PR in favour of https://github.com/ansible/ansible/pull/40575

achinthagunasekara commented:
Closing this PR in favour of https://github.com/ansible/ansible/pull/40575
